ssl/t1_lib.c

```diff
@@ -525,19 +525,33 @@ static int add_provider_sigalgs(const OSSL_PARAM params[], void *data)
             goto err;
     }
 
-    /* The remaining parameters below are mandatory again */
-    p = OSSL_PARAM_locate_const(params, OSSL_CAPABILITY_TLS_SIGALG_MIN_TLS);
-    if (p == NULL || !OSSL_PARAM_get_int(p, &sinf->mintls)) {
+    /* Optional, not documented prior to 3.5 */
+    sinf->mindtls = sinf->maxdtls = -1;
+    p = OSSL_PARAM_locate_const(params, OSSL_CAPABILITY_TLS_SIGALG_MIN_DTLS);
+    if (p != NULL && !OSSL_PARAM_get_int(p, &sinf->mindtls)) {
         ERR_raise(ERR_LIB_SSL, ERR_R_PASSED_INVALID_ARGUMENT);
         goto err;
     }
-    if ((sinf->mintls != 0) && (sinf->mintls != -1) &&
-        ((sinf->mintls < TLS1_3_VERSION))) {
-        /* ignore this sigalg as this OpenSSL doesn't know how to handle it */
-        ret = 1;
+    p = OSSL_PARAM_locate_const(params, OSSL_CAPABILITY_TLS_SIGALG_MAX_DTLS);
+    if (p != NULL && !OSSL_PARAM_get_int(p, &sinf->maxdtls)) {
+        ERR_raise(ERR_LIB_SSL, ERR_R_PASSED_INVALID_ARGUMENT);
+        goto err;
+    }
+    /* DTLS version numbers grow downward */
+    if ((sinf->maxdtls != 0) && (sinf->maxdtls != -1) &&
+        ((sinf->maxdtls > sinf->mindtls))) {
+        ERR_raise(ERR_LIB_SSL, ERR_R_PASSED_INVALID_ARGUMENT);
         goto err;
     }
+    /* No provider sigalgs are supported in DTLS, reset after checking. */
+    sinf->mindtls = sinf->maxdtls = -1;
```

Comment:
This seems to imply the second meaning is correct regarding documentation, right? fwiw, this will need updating should DTLS 1.3 land (see https://github.com/open-quantum-safe/oqs-provider/pull/586) -> Worthwhile adding a "TODO (DTLS1_3)" in the comment to not be surprised if this code merges into that feature branch?

Reply:
Yes, this will be pretty clear once DTLS 1.3 lands.